ui/src/app/components/apps-panel/expanded-app.tsx

```diff
@@ -160,18 +166,24 @@ const CromwellButtonRow = (props: {
   );
 };
 
-const RStudioButtonRow = (props: {
+interface rstudioProps extends NavigationProps {
   userApp: UserAppEnvironment;
-  workspaceNamespace: string;
-}) => {
-  const { userApp, workspaceNamespace } = props;
-
+  workspace: WorkspaceData;
+}
+const RStudioButtonRow = fp.flow(
+  withNavigation,
+  withCurrentWorkspace()
+)((props: rstudioProps) => {
+  const {
+    userApp,
+    workspace: { namespace, id },
+    navigate,
+  } = props;
   const onClickLaunch = async () => {
-    openRStudio(workspaceNamespace, userApp);
+    openRStudioOrConfigPanel(namespace, id, [userApp], 'default', navigate);
   };
-
+  const workspaceNamespace = props.workspace.namespace;
```

Comment:
true i could have left it as is but the pause component is being called as 

<PauseUserAppButton {...{ userApp, workspaceNamespace }} /> <---- L194
i can change the passing variable in PauseUserAppButton OR 
move before L183 
const workspaceNamespace = props.workspace.namespace;
And use the same variable to callopenRStudio

Reply:
It's generally good practice to avoid having two variables with the same value, to avoid developer confusion and prevent the possibility that they will get out of sync.  Thanks for fixing this.